pilot/pkg/config/kube/gateway/deploymentcontroller.go

```diff
@@ -663,6 +663,17 @@ func extractServicePorts(gw gateway.Gateway) []corev1.ServicePort {
 	return svcPorts
 }
 
+// ListenerName allows periods and 253 chars.
+// We map this to service port name which does not allow period and only 63 chars.
+func sanitizeListenerNameForPort(s string) string {
+	// In theory, this mapping can result in a duplicate, but probably not likely
+	s = strings.ReplaceAll(s, ".", "-")
+	if len(s) <= 63 {
+		return s
+	}
+	return s[:63]
```

Comment:
There can be a case generating duplicate svc port name, which is not allowed by k8s